module/updatable_configs/manager.go

```diff
@@ -0,0 +1,217 @@
+package updatable_configs
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+)
+
+// ErrAlreadyRegistered is returned when a config field is registered with a name
+// conflicting with an already registered config field.
+var ErrAlreadyRegistered = fmt.Errorf("config name already registered")
+
+// ValidationError is returned by a config setter function (Set*ConfigFunc) when
+// the provided config field is invalid, and was not applied.
+type ValidationError struct {
+	Err error
+}
+
+func (err ValidationError) Error() string {
+	return err.Err.Error()
+}
+
+func NewValidationErrorf(msg string, args ...any) ValidationError {
+	return ValidationError{
+		Err: fmt.Errorf(msg, args...),
+	}
+}
+
+func IsValidationError(err error) bool {
+	return errors.As(err, &ValidationError{})
+}
+
+type (
+	SetAnyConfigFunc func(any) error
+	GetAnyConfigFunc func() any
+)
+
+// The below are typed setter and getter functions for different config types.
+//
+// ADDING A NEW TYPE:
+// If you need to add a new configurable config field with a type not below:
+//  1. Add a new setter and getter type below
+//  2. Add a Register*Config method to the Registerer interface and Manager implementation below
+//  3. Add a TestManager_Register*Config test to the manager_test.go file.
+
+type (
+	// Set*ConfigFunc is a setter function for a single updatable config field.
+	// Returns ValidationError if the new config value is invalid.
+
+	SetUintConfigFunc     func(uint) error
```

Comment:
```suggestion
	SetUintConfigFunc     func(uint) (uint, error)
```
It's better to return a uint as old value, would be useful for logging. So that we log when it's changed from a certain old value to a new value. 

Does generic support type restriction? Meaning restricting both type T here to be the same type.
```
	SetAnyConfigFunc func(T) (T, error)
```